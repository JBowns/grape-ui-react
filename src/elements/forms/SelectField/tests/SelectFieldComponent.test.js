import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { SelectFieldComponent } from '..';
import 'jest-styled-components';

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};
const colorOptions = [
  { color: '#FF5630', label: 'Red', value: 'red' },
  { color: '#FFC400', label: 'Yellow', value: 'yellow' },
  { color: '#36B37E', label: 'Green', value: 'green' },
];

describe('SelectFieldComponent base', () => {
  it('should return a SelectFieldComponent object', () => {
    const element = <ThemeProvider theme={{}}><SelectFieldComponent defaultValue={colorOptions[1]} id="exampleColor" options={colorOptions} /></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});

describe('SelectFieldComponent with some style relate', () => {
  it('should return a SelectFieldComponent lg object', () => {
    const element = <ThemeProvider theme={{}}><SelectFieldComponent defaultValue={colorOptions[1]} id="exampleColor" lg options={colorOptions} /></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a SelectFieldComponent sm object', () => {
    const element = <ThemeProvider theme={{}}><SelectFieldComponent defaultValue={colorOptions[1]} id="exampleColor" options={colorOptions} sm /></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
