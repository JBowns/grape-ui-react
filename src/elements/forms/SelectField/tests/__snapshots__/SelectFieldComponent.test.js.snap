// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldComponent base should return a SelectFieldComponent object 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 4px;
  border-radius: 4px;
  outline: 0;
  width: 100%;
  box-sizing: border-box;
  padding: 1rem;
}

.c0 .grape-ui-select__control,
.c0 .DayPickerInput > input {
  margin: -1rem;
  padding: 1rem;
}

.c0 .grape-ui-select__control + .DayPickerInput-OverlayWrapper,
.c0 .DayPickerInput > input + .DayPickerInput-OverlayWrapper {
  -webkit-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  -ms-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  transform: translateX(-1rem) translateY(calc(1rem + 2px));
}

.c0[disabled] {
  opacity: 0.3;
}

.c0[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c0:focus,
.c0:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c0:focus + label,
.c0:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c0::-webkit-input-placeholder,
.c0 input::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::-moz-placeholder,
.c0 input::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c0:-ms-input-placeholder,
.c0 input:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::placeholder,
.c0 input::placeholder {
  color: hsl(0,0%,75%);
}

.c0 .DayPickerInput > input {
  background: transparent;
  border: 0;
  font: inherit;
  outline: 0;
  width: 100%;
}

<div
  className="c0 css-2b097c-container"
  id="exampleColor"
  onKeyDown={[Function]}
>
  <div
    className="grape-ui-select__control css-6bcw64-Control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="grape-ui-select__value-container grape-ui-select__value-container--has-value css-1jw5w5s-ValueContainer"
    >
      <div
        className="grape-ui-select__single-value css-1y5ucjk-SingleValue"
      >
        Yellow
      </div>
      <div
        className="css-17yls17-Input"
      >
        <div
          className="grape-ui-select__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="grape-ui-select__indicators css-zededq-IndicatorsContainer"
    >
      <span
        className="grape-ui-select__indicator-separator css-1xjgjl1-IndicatorSeparator"
      />
      <div
        aria-hidden="true"
        className="grape-ui-select__indicator grape-ui-select__dropdown-indicator css-vv1fq9-indicatorContainer"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-6q0nyr-Svg"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectFieldComponent with some style relate should return a SelectFieldComponent lg object 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 8px;
  border-radius: 8px;
  outline: 0;
  width: 100%;
  box-sizing: border-box;
  padding: 1rem;
}

.c0 .grape-ui-select__control,
.c0 .DayPickerInput > input {
  margin: -1rem;
  padding: 1rem;
}

.c0 .grape-ui-select__control + .DayPickerInput-OverlayWrapper,
.c0 .DayPickerInput > input + .DayPickerInput-OverlayWrapper {
  -webkit-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  -ms-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  transform: translateX(-1rem) translateY(calc(1rem + 2px));
}

.c0[disabled] {
  opacity: 0.3;
}

.c0[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c0:focus,
.c0:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c0:focus + label,
.c0:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c0::-webkit-input-placeholder,
.c0 input::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::-moz-placeholder,
.c0 input::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c0:-ms-input-placeholder,
.c0 input:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::placeholder,
.c0 input::placeholder {
  color: hsl(0,0%,75%);
}

.c0 .DayPickerInput > input {
  background: transparent;
  border: 0;
  font: inherit;
  outline: 0;
  width: 100%;
}

<div
  className="c0 css-2b097c-container"
  id="exampleColor"
  onKeyDown={[Function]}
>
  <div
    className="grape-ui-select__control css-6bcw64-Control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="grape-ui-select__value-container grape-ui-select__value-container--has-value css-1jw5w5s-ValueContainer"
    >
      <div
        className="grape-ui-select__single-value css-1y5ucjk-SingleValue"
      >
        Yellow
      </div>
      <div
        className="css-17yls17-Input"
      >
        <div
          className="grape-ui-select__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-3-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="grape-ui-select__indicators css-zededq-IndicatorsContainer"
    >
      <span
        className="grape-ui-select__indicator-separator css-1xjgjl1-IndicatorSeparator"
      />
      <div
        aria-hidden="true"
        className="grape-ui-select__indicator grape-ui-select__dropdown-indicator css-vv1fq9-indicatorContainer"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-6q0nyr-Svg"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectFieldComponent with some style relate should return a SelectFieldComponent sm object 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  box-sizing: border-box;
  padding: 1rem;
}

.c0 .grape-ui-select__control,
.c0 .DayPickerInput > input {
  margin: -1rem;
  padding: 1rem;
}

.c0 .grape-ui-select__control + .DayPickerInput-OverlayWrapper,
.c0 .DayPickerInput > input + .DayPickerInput-OverlayWrapper {
  -webkit-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  -ms-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  transform: translateX(-1rem) translateY(calc(1rem + 2px));
}

.c0[disabled] {
  opacity: 0.3;
}

.c0[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c0:focus,
.c0:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c0:focus + label,
.c0:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c0::-webkit-input-placeholder,
.c0 input::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::-moz-placeholder,
.c0 input::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c0:-ms-input-placeholder,
.c0 input:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::placeholder,
.c0 input::placeholder {
  color: hsl(0,0%,75%);
}

.c0 .DayPickerInput > input {
  background: transparent;
  border: 0;
  font: inherit;
  outline: 0;
  width: 100%;
}

<div
  className="c0 css-2b097c-container"
  id="exampleColor"
  onKeyDown={[Function]}
>
  <div
    className="grape-ui-select__control css-6bcw64-Control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="grape-ui-select__value-container grape-ui-select__value-container--has-value css-1jw5w5s-ValueContainer"
    >
      <div
        className="grape-ui-select__single-value css-1y5ucjk-SingleValue"
      >
        Yellow
      </div>
      <div
        className="css-17yls17-Input"
      >
        <div
          className="grape-ui-select__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-4-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="grape-ui-select__indicators css-zededq-IndicatorsContainer"
    >
      <span
        className="grape-ui-select__indicator-separator css-1xjgjl1-IndicatorSeparator"
      />
      <div
        aria-hidden="true"
        className="grape-ui-select__indicator grape-ui-select__dropdown-indicator css-vv1fq9-indicatorContainer"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-6q0nyr-Svg"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
